### PR TITLE
[tcp] style changes, typo fixes, and smaller changes

### PR DESCRIPTION
**Pull request by abtink**
This commit contains smaller style/code changes in `tcp6` modules.

### CODE REVIEW

**Review by simonlingoogle — APPROVED** [review]
LGTM 👍

**Review by samkumar — APPROVED** [review]
Looks good. Thanks @abtink for the careful read.

File: src/core/net/tcp6.hpp [hunk 1]
```diff
@@ -289,14 +301,15 @@ class Tcp : public InstanceLocator, private NonCopyable
...
+         *
          */
         Error Abort(void);
 
         /**
          * Deinitializes this TCP endpoint.
          *
          * This means that OpenThread no longer keeps track of this TCP endpoint and
-         * deallocates all resources it has internally allocated for this TCP endpoint.
+         * de-allocates all resources it has internally allocated for this TCP endpoint.
```
Comment (jwhui) [inline src/core/net/tcp6.hpp:312]: I believe `deallocates` is accepted as a word: https://www.google.com/search?q=deallocates
```suggestion
         * deallocates all resources it has internally allocated for this TCP endpoint.
```

File: src/core/net/tcp6.hpp [hunk 2]
```diff
@@ -381,14 +397,15 @@ class Tcp : public InstanceLocator, private NonCopyable
...
+         *
          */
         Error StopListening(void);
 
         /**
          * Deinitializes this TCP listener.
          *
          * This means that OpenThread no longer keeps track of this TCP listener and
-         * deallocates all resources it has internally allocated for this TCP listener.
+         * de-allocates all resources it has internally allocated for this TCP listener.
```
Comment (jwhui) [inline src/core/net/tcp6.hpp:408]: I believe `deallocates` is accepted as a word: https://www.google.com/search?q=deallocates
```suggestion
         * deallocates all resources it has internally allocated for this TCP endpoint.
```

**Review by jwhui — APPROVED** [review]

**Pull request merged**